src/datadog/config_manager.cpp

```diff
@@ -42,7 +43,10 @@ Expected<Rules> parse_trace_sampling_rules(const nlohmann::json& json_rules) {
         return *error;
       }
 
-      rate.value = *maybe_rate;
+      rule.rate = *maybe_rate;
+    } else {
+      return Error{Error::TRACE_SAMPLING_RULES_INVALID_JSON,
+                   "Missing \"sample_rate\" field"};
```

Comment:
qq for this and other new error handling: Is a default value set somewhere? Or will parsing now fail when it didn't before?

Reply:
Parsing will fail, those are mandatory fields.